File: jvm-runtime/testdata/java/javaserver/src/main/java/xyz/block/ftl/javacomments/CustomSerializedTypeMapper.java
package xyz.block.ftl.test;

import xyz.block.ftl.TypeAlias;
import xyz.block.ftl.TypeAliasMapper;

/**
 * Comment on a TypeAlias
 */
@TypeAlias(name = "CustomSerializedType")
public class CustomSerializedTypeMapper implements TypeAliasMapper<xyz.block.ftl.test.CustomSerializedType, String> {
    @Override
    public String encode(xyz.block.ftl.test.CustomSerializedType object) {
        return object.getValue();
    }

    @Override
    public xyz.block.ftl.test.CustomSerializedType decode(String serialized) {
        return new xyz.block.ftl.test.CustomSerializedType(serialized);
    }
}

File: jvm-runtime/testdata/kotlin/kotlinmodule/src/main/kotlin/xyz/block/ftl/test/CustomSerializedTypeMapper.kt
package xyz.block.ftl.test

import xyz.block.ftl.TypeAlias
import xyz.block.ftl.TypeAliasMapper

@TypeAlias(name = "CustomSerializedType")
class CustomSerializedTypeMapper : TypeAliasMapper<CustomSerializedType, String> {
    override fun encode(`object`: CustomSerializedType): String {
        return `object`.value
    }

    override fun decode(serialized: String): CustomSerializedType {
        return CustomSerializedType(serialized)
    }
}
